### PR TITLE
Rename docker folder to containers

### PR DESCRIPTION
**Pull request by aesteve-rh**
Rename docker folder to containers to make
it more generic. Dockerfiles contained in
this folder are also renamed to containerfile.

Update the development document to reflect
the new path.

Fixes: https://github.com/oVirt/ovirt-imageio/issues/9
Signed-off-by: Albert Esteve <aesteve@redhat.com>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,14 +20,14 @@ jobs:
     - name: Configure docker
       run: sudo ./ci/configure-docker.py
     - name: Pull container
-      run: docker pull quay.io/ovirt/imageio-test-${{matrix.distro}}
+      run: docker pull quay.io/alesgar/imageio-test-${{matrix.distro}}
```
Comment (nirs) [inline .github/workflows/ci.yml:23]: This is temporary change for testing? The PR should not include these changes. You can
make this in a private branch in your fork for testing, and share a link to the build using these changes, but please avoid sending temporary changes to PRs.

**Review by nirs — CHANGES_REQUESTED** [review]
The renames and changes to the makfile looks good, but the temporary ci changes should not be included in a PR, and the tests fail.

**Comment by nirs** [comment]
Tests failures seems random (centos-8 succeeded, other failed):

```
2022-07-09 14:53:39,108 DEBUG   (MainThread) [services] Creating local.service on socket ''
2022-07-09 14:53:39,109 ERROR   (MainThread) [server] Server failed
Traceback (most recent call last):
  File "/src/ovirt_imageio/_internal/server.py", line 47, in main
    server = Server(cfg)
  File "/src/ovirt_imageio/_internal/server.py", line 124, in __init__
    self.local_service = services.LocalService(self.config, self.auth)
  File "/src/ovirt_imageio/_internal/services.py", line 123, in __init__
    self._server = uhttp.Server(config.local.socket, uhttp.Connection)
  File "/src/ovirt_imageio/_internal/http.py", line 99, in __init__
    self.server_bind()
  File "/src/ovirt_imageio/_internal/uhttp.py", line 75, in server_bind
    self.socket.bind(self.server_address)
OSError: [Errno 98] Address already in use
```

socket '' looks wrong, the unix socket cannot be empty.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (aesteve-rh) [reply .github/workflows/ci.yml:23]: The idea was to test it and then remove the commit and leave it ready. But it failed late last Friday, so I left it in.

I'll separate the commit to my fork, and take it into account for future image updates.

**Review by nirs — APPROVED** [review]

**Pull request merged**